bundle-workflow/src/run_integ_test.py

```diff
@@ -16,56 +15,19 @@
 from manifests.test_manifest import TestManifest
 from system import console
 from system.temporary_directory import TemporaryDirectory
+from test_workflow.dependency_installer import DependencyInstaller
 from test_workflow.integ_test.integ_test_suite import IntegTestSuite
 from test_workflow.test_args import TestArgs
 
-# TODO: 1. log test related logging into a log file. Output only workflow logs on shell.
-# TODO: 2. Move common functions to utils.py
 
-COMMON_DEPENDENCIES = ["OpenSearch", "common-utils", "job-scheduler", "alerting"]
-
-
-# TODO: replace with DependencyProvider - https://github.com/opensearch-project/opensearch-build/issues/283
-def pull_common_dependencies(work_dir, build_manifest):
-    logging.info("Pulling common dependencies for integration tests")
+def pull_build_repo(work_dir):
     logging.info("Pulling opensearch-build")
     os.chdir(work_dir)
```

Comment:
The `chdir` feels like an undesirable side-effect, `GitRepository` certainly doesn't need it and you already do it in the calling code.